### PR TITLE
Support for ON_REQUESTED_INCOMPATIBLE_QOS and ON_OFFERED_INCOMPATIBLE_QOS events

### PR DESCRIPTION
**Pull request by mm318**
Related to [this](https://github.com/ros2/ros2/issues/822) feature request. The design and implementation details can also be found there.

Handling events of type `REQUESTED_INCOMPATIBLE_QOS_STATUS` and `OFFERED_INCOMPATIBLE_QOS_STATUS`

### CODE REVIEW

File: rmw_connext_cpp/src/connext_static_publisher_info.cpp [hunk 1]
```diff
@@ -56,6 +57,28 @@ rmw_ret_t ConnextStaticPublisherInfo::get_status(
...
+        break;
+      }
+    case DDS::StatusKind::DDS_OFFERED_INCOMPATIBLE_QOS_STATUS:
+      {
+        DDS::OfferedIncompatibleQosStatus offered_incompatible_qos;
+        DDS::ReturnCode_t dds_return_code =
+          topic_writer_->get_offered_incompatible_qos_status(offered_incompatible_qos);
+
+        rmw_ret_t from_dds = check_dds_ret_code(dds_return_code);
+        std::cout << "from_dds: " << from_dds << std::endl;
```
Comment (ivanpauno) [inline rmw_connext_cpp/src/connext_static_publisher_info.cpp]: leftover from debugging?

File: rmw_connext_cpp/src/connext_static_publisher_info.cpp [hunk 2]
```diff
@@ -56,6 +57,28 @@ rmw_ret_t ConnextStaticPublisherInfo::get_status(
...
+        if (from_dds != RMW_RET_OK) {
```
Comment (ivanpauno) [inline rmw_connext_cpp/src/connext_static_publisher_info.cpp]: nit: should be `RMW_RET_OK != from_dds` according to our style guide.

File: rmw_connext_cpp/src/connext_static_subscriber_info.cpp [hunk 3]
```diff
@@ -58,6 +59,28 @@ rmw_ret_t ConnextStaticSubscriberInfo::get_status(
...
+        break;
+      }
+    case DDS::StatusKind::DDS_REQUESTED_INCOMPATIBLE_QOS_STATUS:
+      {
+        DDS::RequestedIncompatibleQosStatus requested_incompatible_qos;
+        DDS::ReturnCode_t dds_return_code =
+          topic_reader_->get_requested_incompatible_qos_status(requested_incompatible_qos);
+
+        rmw_ret_t from_dds = check_dds_ret_code(dds_return_code);
+        std::cout << "from_dds: " << from_dds << std::endl;
```
Comment (ivanpauno) [inline rmw_connext_cpp/src/connext_static_subscriber_info.cpp]: same question here

File: rmw_connext_cpp/src/connext_static_publisher_info.cpp (hunk 1 above)
Comment (mm318) [reply rmw_connext_cpp/src/connext_static_publisher_info.cpp]: Removed. Thanks for catching this!

File: rmw_connext_cpp/src/connext_static_subscriber_info.cpp (hunk 3 above)
Comment (mm318) [reply rmw_connext_cpp/src/connext_static_subscriber_info.cpp]: Removed. Thanks for catching this!

File: rmw_connext_cpp/src/connext_static_publisher_info.cpp (hunk 2 above)
Comment (mm318) [reply rmw_connext_cpp/src/connext_static_publisher_info.cpp]: Done.

**Pull request merged**